python/lsst/daf/ingest/ingestCatalogTask.py

```diff
@@ -0,0 +1,576 @@
+                --host lsst-db.ncsa.illinois.edu \
+                --database $USER_test \
+                --table Source \
+                --dstype src \
+                --id filter=g
+    \endcode
+    """
+
+    ConfigClass = IngestCatalogConfig
+    _DefaultName = "ingestCatalog"
+    RunnerClass = IngestCatalogRunner
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        """Extend the default argument parser with database-specific
+        arguments and the dataset type for the catalogs to be read.
+        """
+        parser = pipeBase.ArgumentParser(name=cls._DefaultName)
+        parser.add_argument(
+            "--host", dest="host", required=True,
+            help="Database hostname")
+        parser.add_argument(
+            "--database", dest="db", required=True,
+            help="Database name")
+        parser.add_argument(
+            "--user", dest="user",
+            help="Database username (optional)", default=None)
+        parser.add_argument(
+            "--port", dest="port",
+            help="Database port number (optional)", default=3306)
+        parser.add_argument(
+            "--table", dest="tableName", required=True,
+            help="Table to ingest into")
+        parser.add_argument(
+            "--view", dest="viewName",
+            help="View to create containing column aliases")
+        parser.add_id_argument(
+            "--id", pipeBase.DatasetArgument("dstype"),
+            help="Dataset data id to ingest")
+            # Use DatasetArgument to require dataset type be specified on
+            # the command line
+        return parser
+
+    def runFile(self, fileName, tableName, host, db, port=3306, user=None, viewName=None):
+        """Ingest a BaseCatalog specified by a filename.
+        """
+        cat = afwTable.BaseCatalog.readFits(fileName)
+        self.ingest(cat, tableName, host, db, port, user, viewName)
+
+    def run(self, dataRef, dstype, tableName, host, db, port=3306, user=None, viewName=None):
+        """Ingest a BaseCatalog specified by a dataref and dataset type.
+        """
+        self.ingest(dataRef.get(dstype), tableName, host, db, port, user, viewName)
+
+    @pipeBase.timeMethod
+    def ingest(self, cat, tableName, host, db, port=3306, user=None, viewName=None):
+        """Ingest a BaseCatalog passed as an object.
+
+        @param cat (BaseCatalog) Catalog to ingest.
+        @param tableName (str)   Name of the database table to create.
+        @param host (str)        Name of the database host machine.
+        @param db (str)          Name of the database to ingest into.
+        @param port (int)        Port number on the database host.
+        @param user (str)        User name to use for the database.
+        @param viewName (str)    Name of the database view to create.
+        """
+        tableName = quoteIdentifier(tableName)
+        viewName = quoteIdentifier(viewName) if viewName else None
```

Comment:
I'm no MySQL expert, but I'm a bit surprised that the constraints on column names (`canonicalizeFieldName()`, above, converts any non-word character to a `_`) are much tighter than those on table & view names (which just need to be quoted).